ert3/evaluator/_evaluator.py

```diff
@@ -110,85 +117,77 @@ def _build_ee_config(
+            .set_mime(output.mime)
+        )
+
+    for input_ in stage.input:
+        step_builder.add_input(
+            create_file_io_builder()
+            .set_name(input_.record)
+            .set_path(Path(input_.location))
+            .set_mime(input_.mime)
         )
 
+    for cmd in commands:
+        step_builder.add_input(
+            create_file_io_builder()
+            .set_name(cmd.name)
+            .set_path(command_location(cmd.name))
+            .set_mime(cmd.mime)
+            .set_executable()
+        )
+
+    if isinstance(stage, ert3.config.Function):
+        step_builder.add_job(
+            create_job_builder()
+            .set_name(stage.function.__name__)
+            .set_executable(cloudpickle.dumps(stage.function))
+        )
     if isinstance(stage, ert3.config.Unix):
         for script in stage.script:
             name, *args = script.split()
-            jobs.append(
-                {
-                    "name": name,
-                    "executable": command_location(name),
-                    "args": tuple(args),
-                }
+            step_builder.add_job(
+                create_job_builder()
+                .set_name(name)
+                .set_executable(command_location(name))
+                .set_args(tuple(args))
             )
 
-    steps = [
-        {
-            "name": stage.name + "-only_step",
-            "inputs": [
-                {
-                    "record": input_.record,
-                    "location": input_.location,
-                    "mime": input_.mime,
-                    "is_executable": False,
-                }
-                for input_ in stage.input
-            ]
-            + [
-                {
-                    "record": cmd.name,
-                    "location": command_location(cmd.name),
-                    "mime": cmd.mime,
-                    "is_executable": True,
-                }
-                for cmd in commands
-            ],
-            "outputs": [
-                {
-                    "record": output.record,
-                    "location": output.location,
-                    "mime": output.mime,
-                }
-                for output in stage.output
-            ],
-            "jobs": jobs,
-            "type": "function" if isinstance(stage, ert3.config.Function) else "unix",
-        }
-    ]
+    builder = (
+        create_ensemble_builder()
+        .set_ensemble_size(ensemble_size)
+        .set_max_running(10000)
+        .set_max_retries(0)
+        .set_executor(ensemble.forward_model.driver)
+        .set_forward_model(
+            create_realization_builder().active(True).add_step(step_builder)
+        )
+    )
 
     inputs = _prepare_input(
         ensemble.storage_type, stage, parameters, storage_path, ensemble_size
     )
     outputs = _prepare_output(ensemble.storage_type, stage, storage_path, ensemble_size)
+    builder = builder.set_inputs(inputs).set_outputs(outputs)
```

Comment:
The following is equivalent, but which do we prefer? :thinking: 

```suggestion
    builder.set_inputs(inputs).set_outputs(outputs)
```

Reply:
Latter? I think

Reply:
I think I agree :+1: